modin/experimental/pandas/test/test_io_exp.py

```diff
@@ -305,14 +306,21 @@ def _custom_parser(io_input, **kwargs):
         df2 = pd.read_json(filename, lines=True)[["col0", "col1", "col3"]].rename(
             columns={"col0": "testID"}
         )
-    df_equals(df1, df2)
+        if ExperimentalAsyncReadMode.get():
+            # If read operations are asynchronous, then the dataframes
+            # check should be inside `ensure_clean_dir` context
+            # because the file may be deleted before actual reading starts
+            df_equals(df1, df2)
+    if not ExperimentalAsyncReadMode.get():
+        df_equals(df1, df2)
```

Comment:
why don't we always compare inside of this context then?
```suggestion
        df_equals(df1, df2)
```

Reply:
To check that nothing will break for users if they do not use new feature.